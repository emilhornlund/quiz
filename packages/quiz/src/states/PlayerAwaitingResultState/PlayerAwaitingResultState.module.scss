@import '../../styles/colors';
@import '../../styles/helpers';

.main {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  @include mobile {
    row-gap: $layout-gutter;
  }

  @include tablet {
    row-gap: $layout-gutter * 1.5;
  }

  @include desktop {
    row-gap: $layout-gutter * 2;
  }

  .title {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    font-weight: bold;
    text-align: center;

    @include mobile {
      @include fontSize(16px);
    }

    @include tablet {
      @include fontSize(46px);
    }

    @include desktop {
      @include fontSize(56px);
    }
  }

  @mixin loading-spinner-dimensions($size) {
    width: $size;
    height: $size;

    $--loading-spinner-bar-left: $size / 10;
    $--loading-spinner-bar-width: $size / 5;

    div {
      left: $--loading-spinner-bar-left;
      width: $--loading-spinner-bar-width;

      @keyframes loading-spinner-animation {
        0% {
          top: $--loading-spinner-bar-left;
          height: $size - ($--loading-spinner-bar-left * 2);
        }
        50%, 100% {
          top: $--loading-spinner-bar-left * 3;
          height: $--loading-spinner-bar-width * 2;
        }
      }

      &:nth-child(1) {
        animation-delay: -0.24s;
        left: $--loading-spinner-bar-left * 1;
      }
      &:nth-child(2) {
        animation-delay: -0.12s;
        left: $--loading-spinner-bar-left * 4;
      }
      &:nth-child(3) {
        animation-delay: 0s;
        left: $--loading-spinner-bar-left * 7;
      }
    }
  }

  .loadingSpinner {
    display: inline-block;
    position: relative;

    @include mobile {
      $--mobile-loading-spinner-size: pxToRem(80px);
      @include loading-spinner-dimensions($--mobile-loading-spinner-size);
    }

    @include tablet {
      $--tablet-loading-spinner-size: pxToRem(100px);
      @include loading-spinner-dimensions($--tablet-loading-spinner-size);
    }

    @include desktop {
      $--desktop-loading-spinner-size: pxToRem(120px);
      @include loading-spinner-dimensions($--desktop-loading-spinner-size);
    }

    div {
      display: inline-block;
      position: absolute;
      background: currentColor;
      animation: loading-spinner-animation 1.2s cubic-bezier(0, 0.5, 0.5, 1) infinite;

      &:nth-child(1) {
        animation-delay: -0.24s;
      }

      &:nth-child(2) {
        animation-delay: -0.12s;
      }

      &:nth-child(3) {
        animation-delay: 0s;
      }
    }
  }
}