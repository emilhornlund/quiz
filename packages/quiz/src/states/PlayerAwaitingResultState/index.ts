export type { PlayerAwaitingResultStateProps } from './PlayerAwaitingResultState'
export { default } from './PlayerAwaitingResultState'
