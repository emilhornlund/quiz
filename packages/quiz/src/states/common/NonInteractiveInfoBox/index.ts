export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export { default } from './NonInteractiveInfoBox'
