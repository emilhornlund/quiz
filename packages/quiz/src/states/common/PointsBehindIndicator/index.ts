export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export { default } from './PointsBehindIndicator'
