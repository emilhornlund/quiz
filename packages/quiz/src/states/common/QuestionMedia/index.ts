export type { QuestionMediaProps } from './QuestionMedia'
export { default } from './QuestionMedia'
