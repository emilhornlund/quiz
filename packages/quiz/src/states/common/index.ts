export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export type { QuestionAnswerPickerProps } from './QuestionAnswerPicker'
export type { QuestionMediaProps } from './QuestionMedia'
export type { QuestionTextPreviewProps } from './QuestionTextPreview'
export type { QuestionTypePointsBarProps } from './QuestionTypePointsBar'

export { default as GamePage } from './GamePage'
export { default as NonInteractiveInfoBox } from './NonInteractiveInfoBox'
export { default as PointsBehindIndicator } from './PointsBehindIndicator'
export { default as QuestionAnswerPicker } from './QuestionAnswerPicker'
export { default as QuestionMedia } from './QuestionMedia'
export { default as QuestionTextPreview } from './QuestionTextPreview'
export { default as QuestionTypePointsBar } from './QuestionTypePointsBar'
