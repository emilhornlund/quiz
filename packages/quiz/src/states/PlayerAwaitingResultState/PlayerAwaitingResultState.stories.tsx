import { GameEventType } from '@quiz/common'
import type { Meta, StoryObj } from '@storybook/react'

import PlayerAwaitingResultState from './PlayerAwaitingResultState'

const meta = {
  component: PlayerAwaitingResultState,
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof PlayerAwaitingResultState>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {
    event: {
      type: GameEventType.GameAwaitingResultPlayer,
      nickname: 'FrostyBear',
      score: {
        total: 10458,
      },
      pagination: {
        current: 1,
        total: 20,
      },
    },
  },
} satisfies Story
