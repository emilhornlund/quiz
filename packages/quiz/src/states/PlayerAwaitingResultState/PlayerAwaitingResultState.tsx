import { GameAwaitingResultPlayerEvent } from '@quiz/common'
import React, { FC } from 'react'

import {
  LoadingSpinner,
  PlayerGameFooter,
  RocketImage,
  Typography,
} from '../../components'
import { GamePage } from '../common'

export interface PlayerAwaitingResultStateProps {
  event: GameAwaitingResultPlayerEvent
}

const PlayerAwaitingResultState: FC<PlayerAwaitingResultStateProps> = ({
  event: {
    player: {
      nickname,
      score: { total: totalScore },
    },
    pagination: { current: currentQuestion, total: totalQuestions },
  },
}) => (
  <GamePage
    footer={
      <PlayerGameFooter
        currentQuestion={currentQuestion}
        totalQuestions={totalQuestions}
        nickname={nickname}
        totalScore={totalScore}
      />
    }>
    <RocketImage />
    <Typography variant="title" size="medium">
      Waiting for other players
    </Typography>
    <LoadingSpinner />
  </GamePage>
)

export default PlayerAwaitingResultState
