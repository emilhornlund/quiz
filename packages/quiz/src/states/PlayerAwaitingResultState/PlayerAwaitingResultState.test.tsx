import { GameEventType } from '@quiz/common'
import { render } from '@testing-library/react'
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { describe, expect, it } from 'vitest'

import PlayerAwaitingResultState from './PlayerAwaitingResultState'

describe('PlayerAwaitingResultState', () => {
  it('should render PlayerAwaitingResultState with default props', async () => {
    const { container } = render(
      <MemoryRouter>
        <PlayerAwaitingResultState
          event={{
            type: GameEventType.GameAwaitingResultPlayer,
            player: {
              nickname: 'FrostyBear',
              score: {
                total: 10458,
              },
            },
            pagination: {
              current: 1,
              total: 20,
            },
          }}
        />
      </MemoryRouter>,
    )

    expect(container).toMatchSnapshot()
  })
})
