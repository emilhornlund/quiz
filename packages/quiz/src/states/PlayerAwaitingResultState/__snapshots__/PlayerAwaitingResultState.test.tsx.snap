// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayerAwaitingResultState > should render PlayerAwaitingResultState with default props 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <div
          class="buttonInputContainer buttonInputKindPrimary buttonInputSizeSmall"
        >
          <button
            data-testid="test-fullscreen-button-button"
            id="fullscreen-button"
            name="fullscreen-button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-maximize "
              data-icon="maximize"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M200 32L56 32C42.7 32 32 42.7 32 56l0 144c0 9.7 5.8 18.5 14.8 22.2s19.3 1.7 26.2-5.2l40-40 79 79-79 79L73 295c-6.9-6.9-17.2-8.9-26.2-5.2S32 302.3 32 312l0 144c0 13.3 10.7 24 24 24l144 0c9.7 0 18.5-5.8 22.2-14.8s1.7-19.3-5.2-26.2l-40-40 79-79 79 79-40 40c-6.9 6.9-8.9 17.2-5.2 26.2s12.5 14.8 22.2 14.8l144 0c13.3 0 24-10.7 24-24l0-144c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2l-40 40-79-79 79-79 40 40c6.9 6.9 17.2 8.9 26.2 5.2s14.8-12.5 14.8-22.2l0-144c0-13.3-10.7-24-24-24L312 32c-9.7 0-18.5 5.8-22.2 14.8s-1.7 19.3 5.2 26.2l40 40-79 79-79-79 40-40c6.9-6.9 8.9-17.2 5.2-26.2S209.7 32 200 32z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="content fullWidth centerAlign"
    >
      <div
        class="main"
      >
        <img
          alt="Hourglass"
          src="data:image/svg+xml,%3csvg%20width='170'%20height='224'%20viewBox='0%200%20170%20224'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M150%200C161.046%200%20170%208.95431%20170%2020C170%2027.8081%20165.525%2034.5695%20159%2037.8633V51C159%2076.3133%20146.29%2098.6573%20126.903%20112C146.29%20125.343%20159%20147.687%20159%20173V186.136C165.525%20189.429%20170%20196.192%20170%20204C170%20215.046%20161.046%20224%20150%20224H20C8.95431%20224%200%20215.046%200%20204C0%20196.192%204.4751%20189.429%2011%20186.136V173C11%20147.687%2023.7098%20125.343%2043.0957%20112C23.7098%2098.6573%2011%2076.313%2011%2051V37.8633C4.47531%2034.5695%200%2027.8081%200%2020C0%208.95431%208.95431%204.02661e-08%2020%200H150ZM20%208C13.3726%208%208%2013.3726%208%2020C8%2026.6274%2013.3726%2032%2020%2032H19V51C19%2078.51%2035.8314%20102.088%2059.7578%20112C35.8314%20121.912%2019%20145.49%2019%20173V192H20C13.3726%20192%208%20197.373%208%20204C8%20210.627%2013.3726%20216%2020%20216H150C156.627%20216%20162%20210.627%20162%20204C162%20197.709%20157.16%20192.549%20151%20192.041V173C151%20145.49%20134.168%20121.912%20110.241%20112C134.168%20102.088%20151%2078.5103%20151%2051V32H150C156.627%2032%20162%2026.6274%20162%2020C162%2013.3726%20156.627%208%20150%208H20Z'%20fill='white'/%3e%3cpath%20d='M151%2032V51C151%2078.5103%20134.168%20102.088%20110.241%20112C134.168%20121.912%20151%20145.49%20151%20173V192H19V173C19%20145.49%2035.8314%20121.912%2059.7578%20112C35.8314%20102.088%2019%2078.51%2019%2051V32H151Z'%20fill='%23FFD32A'/%3e%3cpath%20d='M150%20192C156.627%20192%20162%20197.373%20162%20204C162%20210.627%20156.627%20216%20150%20216H20C13.3726%20216%208%20210.627%208%20204C8%20197.373%2013.3726%20192%2020%20192H150ZM150%208C156.627%208%20162%2013.3726%20162%2020C162%2026.6274%20156.627%2032%20150%2032H20C13.3726%2032%208%2026.6274%208%2020C8%2013.3726%2013.3726%208%2020%208H150Z'%20fill='%233C40C6'/%3e%3cpath%20d='M120.732%2060C126.5%2060.0002%20133.5%2064.0002%20125.5%2072C117.5%2080%2089.5%2089%2089%20105.5C88.9642%20106.681%2088.9333%20107.624%2088.9072%20108.368C88.956%20109.495%2089%20113.118%2089%20122C89%20152%20136.5%20142.5%20136.5%20192H33.5C33.5%20142.5%2081%20152%2081%20122C81%20113.118%2081.044%20109.495%2081.0928%20108.368C81.0667%20107.624%2081.0358%20106.681%2081%20105.5C80.5%2089%2052.5%2080%2044.5%2072C36.5002%2064.0002%2043.4997%2060.0002%2049.2676%2060H120.732Z'%20fill='%23F53B57'/%3e%3c/svg%3e"
        />
      </div>
      <div
        class="typography title medium"
      >
        Waiting for other players
      </div>
      <div
        class="loadingSpinnerContainer"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-question "
            data-icon="circle-question"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
              fill="currentColor"
            />
          </svg>
          <span>
            1
             / 
            20
          </span>
        </div>
        <div
          class="nickname"
        >
          <span>
            FrostyBear
          </span>
        </div>
        <div
          class="score"
        >
          <span>
            10458
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
